client.go

```diff
@@ -12,13 +12,26 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
-func Post(reqUrl string, auth string, body interface{}, contentType string) (respBodyObj ResponseBody, err error) {
+func Post(reqUrl string, auth string, body interface{}, contentType string, bufferBody *bytes.Buffer) (respBodyObj ResponseBody, err error) {
 	postBody, _ := json.Marshal(body)
 	requestBody := bytes.NewBuffer(postBody)
+	var req *http.Request
+
 	log.WithFields(log.Fields{
 		"body": string(postBody),
 	}).Debug("The request body")
-	req, err := http.NewRequest("POST", reqUrl, requestBody)
+	if bufferBody != nil {
```

Comment:
This `if` `else` can be simplified as follows:

```
requestBody := bytes.NewBuffer(postBody)
if bufferBody != nil {
  requestBody = bufferBody
} 
request, reqError := http.NewRequest("POST", reqUrl,  requestBody)
```
The request per-se doesn't change only one param is different.